### ISSUE TITLE
Problem with a linker

### ISSUE THREAD

**Issue by NippleOfAnApe** [djpohly/dwl#360]
dwl: v0.4-rc2-16-g98c7adf (wlroots-next branch)
wlroots: 0.16.0-191-g7f6d646e

I am trying to compile dwl using latest stable wlroots version on git. I get all those undefined references when running make. What could be the problem? 

```
c99 dwl.o util.o `pkg-config --libs wlroots wayland-server xkbcommon libinput `   `pkg-config --cflags wlroots wayland-server xkbcommon libinput ` -I. -DWLR_USE_UNSTABLE -D_POSIX_C_SOURCE=200809L -DVERSION=\"`git describe --long --tags
--dirty 2>/dev/null || echo 0.4-rc2`\"  -pedantic -Wall -Wextra -Wdeclaration-after-statement -Wno-unused-parameter -Wno-sign-compare -Wshadow -Wunused-macros -Werror=strict-prototypes -Werror=implicit -Werror=return-type -Werror=incompatible-pointer-types -O1 -o dwl
/bin/ld: dwl.o: in function `resize':
dwl.c:(.text+0x2b1): undefined reference to `wlr_xdg_toplevel_set_bounds'
/bin/ld: dwl.o: in function `checkidleinhibitor':
dwl.c:(.text+0x6ed): undefined reference to `wlr_idle_notifier_v1_set_inhibited'
/bin/ld: dwl.o: in function `xytonode':
dwl.c:(.text+0x7f3): undefined reference to `wlr_scene_buffer_from_node'
/bin/ld: dwl.c:(.text+0x7fb): undefined reference to `wlr_scene_surface_from_buffer'
/bin/ld: dwl.o: in function `arrangelayer':
dwl.c:(.text+0xb6a): undefined reference to `wlr_scene_layer_surface_v1_configure'
/bin/ld: dwl.o: in function `keypress':
dwl.c:(.text+0xe66): undefined reference to `wlr_idle_notifier_v1_notify_activity'
/bin/ld: dwl.o: in function `inputdevice':
dwl.c:(.text+0x1160): undefined reference to `wlr_keyboard_from_input_device'
/bin/ld: dwl.c:(.text+0x1172): undefined reference to `wlr_pointer_from_input_device'
/bin/ld: dwl.o: in function `axisnotify':
dwl.c:(.text+0x1341): undefined reference to `wlr_idle_notifier_v1_notify_activity'
/bin/ld: dwl.o: in function `createlocksurface':
dwl.c:(.text+0x16ed): undefined reference to `wlr_session_lock_surface_v1_configure'
/bin/ld: dwl.o: in function `motionnotify':
dwl.c:(.text+0x1c45): undefined reference to `wlr_idle_notifier_v1_notify_activity'
/bin/ld: dwl.o: in function `locksession':
dwl.c:(.text+0x251b): undefined reference to `wlr_session_lock_v1_destroy'
/bin/ld: dwl.c:(.text+0x25bc): undefined reference to `wlr_session_lock_v1_send_locked'
/bin/ld: dwl.o: in function `createlayersurface':
dwl.c:(.text+0x36ff): undefined reference to `wlr_scene_layer_surface_v1_create'
/bin/ld: dwl.o: in function `updatemons':
dwl.c:(.text+0x3ad2): undefined reference to `wlr_session_lock_surface_v1_configure'
/bin/ld: dwl.o: in function `requestmonstate':
dwl.c:(.text+0x3d53): undefined reference to `wlr_output_commit_state'
/bin/ld: dwl.o: in function `startdrag':
dwl.c:(.text+0x3f23): undefined reference to `wlr_scene_drag_icon_create'
/bin/ld: dwl.o: in function `buttonpress':
dwl.c:(.text+0x3f8a): undefined reference to `wlr_idle_notifier_v1_notify_activity'
/bin/ld: dwl.o: in function `main':
dwl.c:(.text+0x48a3): undefined reference to `wlr_single_pixel_buffer_manager_v1_create'
/bin/ld: dwl.c:(.text+0x48af): undefined reference to `wlr_subcompositor_create'
/bin/ld: dwl.c:(.text+0x4954): undefined reference to `wlr_idle_notifier_v1_create'
/bin/ld: dwl.c:(.text+0x49df): undefined reference to `wlr_session_lock_manager_v1_create'
collect2: error: ld returned 1 exit status
make: *** [Makefile:18: dwl] Error 1
```

**Comment by funymeleehao** [comment]
me too

**Comment by sevz17** [comment]
I am sure all those functions exists on wlroots git (aka 0.17-dev)

**Comment by NippleOfAnApe** [comment]
I managed to fix it. (I think) the problem was that my distro's package manager had both wlroots and wlroots-devel packages installed. Shared libraries were in /usr/lib and the ones generated when compiling wlroots from source were in /usr/local/lib. So linker was pointing to /usr/lib (installed with package manager) and not to /usr/local/lib (installed from gitlab). I uninstalled both wlroots and devel version, and exported local library path.

`export LD_LIBRARY_PATH=/lib:/usr/lib:/usr/local/lib`

**Comment by NippleOfAnApe** [comment]
I managed to fix it. (I think) the problem was that my distro's package manager had both wlroots and wlroots-devel packages installed. Shared libraries were in /usr/lib and the ones generated when compiling wlroots from source were in /usr/local/lib. So linker was pointing to /usr/lib (installed with package manager) and not to /usr/local/lib (installed from gitlab). I uninstalled both wlroots and devel version, and exported local library path.

`export LD_LIBRARY_PATH=/lib:/usr/lib:/usr/local/lib`
